phones/models.py

```diff
@@ -231,12 +239,20 @@ def save(self, *args, **kwargs):
         # as realphone
         self.country_code = realphone.country_code.upper()
 
+        # Add US numbers to the Relay messaging service, so it goes into our
+        # US A2P 10DLC campaign
         if self.country_code == "US":
-            # Also add this number to the Relay messaging service, so it goes
-            # into our US A2P 10DLC campaign
-            client.messaging.v1.services(
-                settings.TWILIO_MESSAGING_SERVICE_SID
-            ).phone_numbers.create(phone_number_sid=twilio_incoming_number.sid)
+            try:
+                client.messaging.v1.services(
+                    settings.TWILIO_MESSAGING_SERVICE_SID
+                ).phone_numbers.create(phone_number_sid=twilio_incoming_number.sid)
+            except TwilioRestException as err:
+                if err.status == 409 and err.code == 21710:
+                    # Ignore "Phone Number is already in the Messaging Service"
+                    # https://www.twilio.com/docs/api/errors/21710
+                    pass
```

Comment:
Nit: after the checks above, this means that we have a de-synced DB problem between our DB and Twilio's. We should log this. Can be a fast follow.

Reply:
I'm not sure I agree that we have a de-sync DB problem. Let's chat over Zoom and capture the next action in a ticket.

Reply:
Twillio allows a phone number to be reserved twice without an error. Twillio raises an error when registering a number with the Messaging Service. However, there's no additional information, so no DB sync issue if a number is already registered. No fix, but a log may be useful. In practice, this should only happen when the Relay Number is being manually added, such as via the Django Admin or a shell session, so it will be rare. A log may be useful to confirm that.

Validation in DRF / Django is not straightforward. Validation at the API level is good for error messages, but if you don't validate at save, there are code paths where invalid data can get in. There is room to try other approaches.